sungc/deglint.py

```diff
@@ -128,41 +129,141 @@ def check_bandNum_exist(self, bandNums, required_bandNums):
                     )
                 )
 
-    def find_file(self, path, keyword):
+    def find_file(self, keyword):
         """
-        find a file in a directory (path) that
+        find a file in a directory (self.meas_dict)
         has a keyword in its filename
 
         Parameters
         ----------
-        path : str
-            Directory to search
-
         keyword : str
             keyword
 
         Returns
         -------
-        filename : None or str
+        filename : str
             filename
-            if None then file with keyword not found
+
+        Raises
+        ------
+        Exception if filename not found
         """
         filename = None
-        for root, dirs, files in os.walk(path):
-            for f in files:
-                if f.find(keyword) != -1:
-                    filename = pjoin(root, f)
+        for key in self.meas_dict.keys():
+            if key.find(keyword) != -1:
+                basename = self.meas_dict[key]["path"]
+                filename = self.group_path.joinpath(basename)
+
+        if not filename:
+            raise Exception(
+                "\nfilename with keyword ({0}) not found".format(keyword)
+            )
 
         return filename
 
-    def get_band_list(self, meas_dict):
+    def get_meas_dict(self, dc_dataset):
+        """
+        Get the measurement dictionary from the datacube dataset.
+
+        Parameters
+        ----------
+        dc_dataset : datacube.model.Dataset
+            Datacube dataset object
+
+        Returns
+        -------
+        meas_dict : dict
+            A dictionary containing band information
+        """
+        meas_dict = None
+        image_key = None
+        meas_key = None
+        for key in dc_dataset.metadata_doc.keys():
+            if key == "image":
+                image_key = key
+            if key == "measurements":
+                meas_key = key
+
+        if meas_key is None:
+            # older Open Data Cube metadata version used for Sentinel-2
+            meas_dict = dc_dataset.metadata_doc[image_key]["bands"]
+        else:
+            # newer version used for Landsat
+            meas_dict = dc_dataset.metadata_doc[meas_key]
```

Comment:
You could compact this by checking the presence of a certain key.

e.g.

```python
if "image" in dc_dataset.metadata_doc:
    meas_dict = dc_dataset.metadata_doc["image"]["bands"]
elif "measurement" in dc_dataset.metadata_doc:
    meas_dict = dc_dataset.metadata_doc["measurement"]
else:
    raise Exception("no image datasets/measurements found")
```

Reply:
Thanks, I'll make this change